src/pages/MerchantGiftCards/MerchantGiftCards.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import {
+  getMerchantGiftCards,
+  getSeller,
+} from '../../utilities/api/interactionManager';
+import Loader from '../../components/Loader/Loader';
+import styles from './styles.module.scss';
+import ErrorPage from '../../components/404Page';
+const FilterableTable = require('react-filterable-table');
+
+const MerchantGiftCards = () => {
+  const params = useHistory();
+  let urlParams = params.location.pathname.match(/\/[^/]+/g) as string[];
+  urlParams = urlParams.map((param) => param.replace('/', ''));
+  const metadata = {
+    sellerId: urlParams[0],
+    secretId: urlParams[2],
+  };
+
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [giftCards, setGiftCards] = useState<any | null>();
+  const [seller, setSeller] = useState<any | null>();
+
+  const renderAmount = (props) => {
+    return Intl.NumberFormat('en-US', {
+      style: 'currency',
+      currency: 'USD',
+      maximumSignificantDigits: 4,
+    }).format(props.value / 100);
+  };
+
+  const renderDate = (props) => {
+    return new Date(props.value).toISOString().substring(0, 10);
+  };
+
+  const fields = [
+    {
+      name: 'seller_gift_card_id',
+      displayName: 'Voucher Code \n 礼品券号码',
+      inputFilterable: true,
+      sortable: true,
+    },
+    {
+      name: 'email',
+      displayName: 'Email \n 电子邮件',
+      inputFilterable: true,
+      sortable: true,
+    },
+    {
+      name: 'value',
+      displayName: 'Original Amount \n 购买金额',
+      inputFilterable: true,
+      sortable: true,
+      render: renderAmount,
+    },
+    {
+      name: 'created_at',
+      displayName: 'Date Purchased \n 购买日期',
+      inputFilterable: true,
+      sortable: true,
+      render: renderDate,
+    },
+    /* {
+      name: 'expiration',
+      displayName: 'Expiration',
+      inputFilterable: true,
+      sortable: true,
+    },
+    Uncomment if we want expiration */
+  ];
+
+  const fetchData = async () => {
+    try {
+      const giftCardData = (
+        await getMerchantGiftCards(metadata.sellerId, metadata.secretId)
+      ).data;
+      const sellerData = (await getSeller(metadata.sellerId)).data;
+
+      setGiftCards(giftCardData);
+      setSeller(sellerData);
+    } catch {
+      setError(true);
+    }
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setLoading(true);
+    fetchData();
+
+    // eslint-disable-next-line
+  }, []);
+
+  return (
+    <>
+      {loading ? (
+        <Loader isPage={true} />
+      ) : error ? (
+        <ErrorPage menuOpen={false} />
+      ) : (
+        <>
+          <div className={styles.header}>
+            <h1>Voucher Tracker <span className={styles.noBreak}>礼品券记录</span></h1>
```

Comment:
im not sure how the design was for this, but maybe we should think about using i18n for these headers? But then again this is MVP so its a good fit for now.

Reply:
I was initially going to use i18n, but the design actually calls for the page to be bilingual, with both on the page at the same time. I think it's spacious enough to work. The `noBreak` here was because the default word-wrapping was cutting in the middle of the chinese, where I think it makes more sense to drop it all to the second line.
https://www.figma.com/file/I0XPeX6re8t14J4nZOvVNp/Send-Chinatown-Love?node-id=4130%3A6669